src/compile/scale.ts

```diff
@@ -43,8 +47,35 @@ export function compileScales(channels: Channel[], model: Model) {
         }
       });
 
-      return scaleDef;
-    });
+      if (channel === COLOR && (fieldDef.type === ORDINAL || fieldDef.bin)) {
+        scales.push({
```

Comment:
Add comment that this scale provides mapping for the rank field back to original data for legend labels.